node_modules/col-assets/tests/test-migrate.js

```diff
@@ -130,11 +133,44 @@ describe('Migrate', function() {
     /**
      * Test that verifies migration from legacy filesystem to Amazon S3
      */
-    it('migrates file from Amazon S3 to Amazon S3', function(callback) {
+    it('migrates files from Amazon S3 to Amazon S3', function(callback) {
+      var course = TestsUtil.generateCourse(global.tests.canvas.ucberkeley);
+      var instructor = TestsUtil.generateInstructor(global.tests.canvas.ucberkeley);
+      var categoryName = 'Category 1';
+
+      // Instructor is teaching course with files in Canvas filesystem
+      AssetsTestUtil.setUpAmazonS3BackedCourse(instructor, function(client1, course1, instructor1) {
+        CategoriesTestUtil.assertCreateCategory(client1, course1, categoryName, function(category) {
+
+          // Put file to Canvas filesystem
+          var opts = {'categories': category.id};
+          AssetsTestUtil.assertFileCreateAndStorage(client1, course1, opts, function(assets) {
+            assert.ok(_.startsWith(assets[0].download_url, 's3://'));
+
+            // Target course has files in Amazon S3
+            AssetsTestUtil.setUpAmazonS3BackedCourse(instructor, function(client2, course2, instructor2) {
+
+              // Migrate assets from instructor1's course to Amazon S3 backed course
+              UsersTestUtil.assertGetMe(client2, course2, null, function(s3me) {
+                AssetsTestUtil.assertMigrationCompletes(client1, course1, s3me.id, 3, 0, function() {
 
-      // TODO
+                  // Get s3Instructor's assets in Amazon S3 back course
```

Comment:
@johncrossman Minor: based on this comment, were you intending to call the second round of variables `s3client` and `s3course` instead of `client2` and `course2`, to make the distinction clearer? I see what's going on but it took me a moment.